public/app/features/datasources/settings/BasicSettings.test.tsx

```diff
@@ -11,13 +11,16 @@ const setup = () => {
     onNameChange: jest.fn(),
   };
 
-  return shallow(<BasicSettings {...props} />);
+  return render(<BasicSettings {...props} />);
 };
 
-describe('Render', () => {
-  it('should render component', () => {
-    const wrapper = setup();
+describe('Basic Settings', () => {
+  it('should render component', async () => {
+    setup();
 
-    expect(wrapper).toMatchSnapshot();
+    expect(screen.getByLabelText('Name')).toBeInTheDocument();
+    expect(screen.getByPlaceholderText('Name')).toBeInTheDocument();
+    expect(screen.getByLabelText('Default')).toBeInTheDocument();
+    expect(screen.getByRole('checkbox')).toBeInTheDocument();
```

Comment:
```suggestion
    expect(screen.getByRole('textbox', { name: selectors.pages.DataSource.name })).toBeInTheDocument();
    expect(screen.getByRole('checkbox', { name: 'Default' })).toBeInTheDocument();
```
It's better to use `*byRole` queries whenever possible, since `*byText` will match any text occurrences on the page, which is often something we don't want to.  We can narrow down the match by providing `name` option, which is the element's [accessible name](https://www.tpgi.com/what-is-an-accessible-name/). 

`selectors` can be imported from `@grafana/e2e-selectors`.

Reply:
Thanks for the feedback! Makes perfect sense, I appreciate the explination.